admin/jobs/__init__.py

```diff
@@ -47,12 +48,177 @@ def create_files():
+                    elif column in ['created', 'updated', 'downloaded', 'completed']:
+                        setattr(record, column, datetime.strptime(a[idx], '%Y-%m-%d  %H:%M:%S.%f'))
+                    else:
+                        val = str(f.decrypt(eval(a[idx])), 'utf-8')
+                        if val == 'True':
+                            val = True
+                        elif val == 'False':
+                            val = False
+                        elif val == 'None':
+                            val = None
+                        if val != None:
+                            setattr(record, column, val)
+
+            db.session.add(record)
+            db.session.commit()
+
+    except Exception as e:
+        print(e, flush=True)
+
+    return ('', 200)
+
+
+@jobs.route('/jobs/backup-files1', methods=['GET'])
+#@JobTokenRequired
+def backup_files1():
+    import io
+    import time
+    import zipfile
+    from grc.external_services.aws_s3_client import AwsS3Client
+
+    tic = time.perf_counter()
+
+    try:
+        awsclient = AwsS3Client()
+        awsclient_external = AwsS3Client(external=True)
+        zip_buffer = io.BytesIO()
+        all_files = awsclient.list_objects()
+
+        with zipfile.ZipFile(zip_buffer, 'x', zipfile.ZIP_DEFLATED, False) as zipper:
+            for object_name in all_files:
+                data = awsclient.download_object(object_name)
+                zipper.writestr(object_name, data.getvalue())
+
+        backup_file = f"{current_app.config['ENVIRONMENT'].lower()}/backups/{datetime.now().strftime('%Y%m%d%H%M')}.zip"
+        awsclient_external.upload_fileobj(zip_buffer, backup_file)
+
+    except Exception as e:
+        print(e, flush=True)
+
+    toc = time.perf_counter()
+    print(f"Finished in {toc - tic:0.4f} seconds", flush=True)
+
+    return ('', 200)
+
+
+@jobs.route('/jobs/backup-files2', methods=['GET'])
+#@JobTokenRequired
+def backup_files2():
+    import time
+    from zipstream import ZipStream
+    import pyAesCrypt
+    from grc.external_services.aws_s3_client import AwsS3Client
+
+    tic = time.perf_counter()
+
+    try:
+        awsclient = AwsS3Client()
+        awsclient_external = AwsS3Client(external=True)
+        all_files = awsclient.list_objects()
+        files = []
+
+        for key in all_files:
+            files.append({'stream': awsclient.stream_download_object(key), 'name': key})
+
+        zip_buffer = ZipStream(files, chunksize=32768)
+
+        fin = io.BytesIO()
+        for data in zip_buffer.stream():
+            fin.write(data)
+        fin.seek(0)
+
+        buffer_size = 64 * 1024
+        secret_key = os.environ.get('SQLALCHEMY_KEY', '')
+        fout = io.BytesIO()
+        pyAesCrypt.encryptStream(fin, fout, secret_key, buffer_size)
+        fout.seek(0)
+
+        backup_file = f"{current_app.config['ENVIRONMENT'].lower()}/backups/{datetime.now().strftime('%Y%m%d%H%M')}.zip.encrypted"
```

Comment:
For this filename:
* `current_app.config['ENVIRONMENT'].lower()` is either "development" or "production".  
  It would be great to use the space name here instead (e.g. sandbox/staging/production).  
  In some other file, I've done this using code like:  
  ```
  space_name = (ConfigHelper.get_vcap_application().space_name.lower()
                if ConfigHelper.get_vcap_application() is not None
                else 'local')
  ```
* It would be nice if the sub-folder was `files` instead of `backups` (currently, we have a `database` folder, so `files` would be a good complement to it - `backups` is a bit ambiguous)
* It would be nice if the date format matched the database backup - just to keep things consistent - e.g. `%Y-%m-%d--%H-%M-%s`

Reply:
Done